extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,182 @@
+        <div class="i-amp-story-quiz-option-container"></div>
+      </div>
+    `;
+
+    this.shadowRoot_ = createShadowRootWithStyle(
+      this.element,
+      shadowElement,
+      CSS
+    );
+
+    // TODO: CONVERT THIS TO STANDARD AMP ERROR REPORTING
+    this.attachContent_(e => {
+      console.log('error!', e);
+      throw new Error(e);
+    });
+    this.attachOptionActionHandlers_();
+
+    // TODO: ATTACH SURFACE OPTIONS USING CSS CUSTOM PROPERTIES
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * @private
+   * Finds the prompt and options content
+   * and adds it to the shadow DOM.
+   *
+   * @param {Function} handleError
+   */
+  attachContent_(handleError) {
+    // TODO: If prompt is optional I need to handle the case where it's not present
+    const prompt = this.element.children[0];
+    // First child must be heading h1-h3
+    if (
+      !(prompt instanceof HTMLHeadingElement) ||
+      prompt.tagName[prompt.tagName.length - 1] > 3
+    ) {
+      handleError(
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+    }
+
+    prompt.setAttribute('class', 'i-amp-story-quiz-prompt');
+
+    const options = Array.from(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      handleError('Improper number of options');
+    }
+
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      handleError('Too many children');
+    }
+  }
+
+  /**
+   * @private
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the shadow DOM.
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   */
+  configureOption_(option, index) {
+    // Transfer the option information into a span -
+    // this allows the option container to house other markup,
+    // such as the answer choices
+    const convertedOption = document.createElement('span');
+    convertedOption.textContent = option.textContent;
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    // Set up the class and add it to the shadow root
+    convertedOption.setAttribute('class', 'i-amp-story-quiz-option');
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-option-container')
+      .appendChild(convertedOption);
+
+    // Create a container for the answer choice and add it to the shadow root
+    const answerChoice = document.createElement('span');
+    answerChoice.textContent = answerChoiceOptions[index];
+    answerChoice.setAttribute('class', 'i-amp-story-quiz-answer-choice');
+    convertedOption.prepend(answerChoice);
```

Comment:
`prepend` doesn't have IE11 support, can you use `insertBefore(answerChoice, firstEl)` instead?